src/signals-mach.c

```diff
@@ -54,7 +54,7 @@ void jl_safepoint_resume_thread_mach(jl_ptls_t ptls2, int16_t tid2)
     for (size_t i = 0; i < suspended_threads.len; i++) {
         uintptr_t item = (uintptr_t)suspended_threads.items[i];
         int16_t tid = (int16_t)item;
-        int8_t gc_state = (int8_t)(item >> 8);
+        int8_t gc_state = (int8_t)(item >> 16);
```

Comment:
Let's do a toplevel `#define` for this and while we're at it, spell it `8*sizeof(int16_t)`.